library/src/loader_impl.cxx

```diff
@@ -191,24 +191,25 @@ void loader_impl::addFile(const std::string& path, bool recursive)
 
   if (vtksys::SystemTools::FileIsDirectory(fullPath))
   {
-    if (recursive)
-    {
-      vtksys::Directory dir;
-      dir.Load(fullPath);
-      std::set<std::string> sortedFiles;
+    vtksys::Directory dir;
+    dir.Load(fullPath);
+    std::set<std::string> sortedFiles;
 
-      // Sorting is necessary as KWSys can provide unsorted files
-      for (unsigned long i = 0; i < dir.GetNumberOfFiles(); i++)
+    // Sorting is necessary as KWSys can provide unsorted files
+    for (unsigned long i = 0; i < dir.GetNumberOfFiles(); i++)
+    {
+      std::string currentFile = dir.GetFile(i);
+      if (currentFile != "." && currentFile != "..")
       {
-        std::string currentFile = dir.GetFile(i);
-        if (currentFile != "." && currentFile != "..")
-        {
-          sortedFiles.insert(currentFile);
-        }
+        sortedFiles.insert(currentFile);
```

Comment:
To check, but I'm pretty sure that preallocating a `std::vector` then sort it after the loop would be more efficient in time and space.

Reply:
It would be, but for such small size it does not matter imo.

Reply:
https://stackoverflow.com/questions/15638024/stdsort-vs-inserting-into-an-stdset